### ISSUE TITLE
Line wrapping after 2011 chars.

### ISSUE THREAD

**Issue by ziofat** [ether/etherpad-lite#3506]
https://github.com/ether/etherpad-lite/blob/develop/src/static/js/contentcollector.js#L718

I found there is a force wrapping for line with more than 2000 chars. But why?

**Comment by muxator** [comment]
Just a little bit of archaeology:

It's always been there since first commit from 2011, I guess it was already there in the original 2008-2009 code from Google:

https://github.com/ether/etherpad-lite/blob/325c322a276d03c7de50233aef63598c7dd8c308/static/js/contentcollector.js#L456

This does not really answer your question, but at least we can safely assume that probably is just technical debt.

**Comment by ziofat** [comment]
I simply removed the limitation, no issues found yet.

**Comment by JohnMcLear** [comment]
Afaik I did test coverage for this back in 2011.  It's due to browser performance issues.  Afaik I spawned a pad w/ like 2k+ chars and the more chars in a line it got exponentially more computationally expensive for the browser to render the pad.  Browsers have changed a LOT since then so it's probably a none issue any more but you would want to add full test coverage before rolling into develop because there is a reason I baked it in.

**Comment by JohnMcLear** [comment]
Tldr; write test coverage

**Comment by ziofat** [comment]
I open this issue because we had a project based on etherpad back in 2018. I removed this limit and it did not raise other issues in my observations.
It's great if you can do a test on it.

**Comment by joassouza** [comment]
Just my 2 cents...
We've been using the limit of 20.000 chars for more than one year without any problem

**Comment by JohnMcLear** [comment]
Okay I tested with large amounts of characters (20k) as line wrapping limit and **I can 100% see a huge drop in responsiveness.**

Chrome and Edge both tested, both have a huge reduction in performance when the line number length is at 20k.

I also uncommented some testing which I made a long time ago to show that this issue is real and that the hard coded number should not be changed.  To run the test visit /tests/frontend/?grep=responsiveness.js - change your number then re-run the test and it will fail with it increased too high.

TLDR; **If you increase the ``lineLimit`` number greater than what it is, you will lose responsiveness when editing.**

I'm closing this without making any changes beyond ensuring we have test coverage.
